Azure/azure-cli
# [Keyvault] `az keyvault set-policy/delete-policy`: Support --application-id

**Description**

Feature requested by #17806 #17156 
Add `--application-id` for `az keyvault set-policy/delete-policy` for compound identity support.

_Related keyvault doc: [identity-management](https://docs.microsoft.com/en-us/azure/key-vault/general/security-features#identity-management)_

**Testing Guide**

set-policy:
`az keyvault set-policy -n testkeyvault --object-id userObjectId --application-id appId --key-permissions get list`
delete-policy:
`az keyvault delete-policy -n testkeyvault --object-id userObjectId --application-id appId`

---

This checklist is used to make sure that common guidelines for a pull request are followed.
